sequence_task_1/student/include/impl.h

```diff
@@ -8,33 +8,85 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    typedef Song_t value_type;
+    typedef Song_t& reference;
+    typedef const Song_t& const_reference;
+    typedef typename Container::iterator iterator;
+    typedef typename Container::const_iterator const_iterator;
+    typedef ptrdiff_t difference_type;
```

Comment:
why do you need to define them manually? I'd stick to the container itself, so we'll add type requirements on our template parameter